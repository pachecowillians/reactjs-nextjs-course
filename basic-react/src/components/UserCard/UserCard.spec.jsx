import { render, screen } from "@testing-library/react";
import { UserCard } from '.'

const props = {
    photo: 'photo.png',
    name: 'name',
    email: 'email',

}

describe('<UserCard />', () => {
    it('should render UserCard correctly', () => {
        // const { debug } = render(<UserCard user={props} />);
        // debug();
        render(<UserCard user={props} />);

        expect(screen.getByRole('img', { name: props.name })).toHaveAttribute('src', props.photo);
        expect(screen.getByRole('heading', { name: props.name })).toBeInTheDocument();
        expect(screen.getByText(props.email)).toBeInTheDocument();
    });

    it('should match snapshot', () => {
        const { container } = render(<UserCard user={props} />);
        expect(container.firstChild).toMatchSnapshot();
    });

});
