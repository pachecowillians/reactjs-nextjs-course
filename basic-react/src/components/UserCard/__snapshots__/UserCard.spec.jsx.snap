// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<UserCard /> should match snapshot 1`] = `
<div
  class="user"
>
  <img
    alt="name"
    src="photo.png"
  />
  <div
    class="user-info"
  >
    <h3>
      name
    </h3>
    <span>
      email
    </span>
  </div>
</div>
`;
